onnx/shape_inference/implementation.h

```diff
@@ -12,6 +12,37 @@
 namespace ONNX_NAMESPACE {
 namespace shape_inference {
 
+struct SymbolTable {
+  SymbolTable(GraphProto* g) : index_(0){
+    existing_symbols.clear();
+    AddExistingSymbolicDims(*g->mutable_input());
+    AddExistingSymbolicDims(*g->mutable_output());
+    AddExistingSymbolicDims(*g->mutable_value_info());
+  }
+  std::string createNew(std::string symbol_prefix="unk__") {
+    std::string newSymbol;
+    do {
+      newSymbol = symbol_prefix + std::to_string(index_++);
+    } while(existing_symbols.count(newSymbol) > 0);
+    return newSymbol;
+  };
+  private:
+    unsigned int index_;
+    std::unordered_set<std::string> existing_symbols;
+    void AddExistingSymbolicDims(google::protobuf::RepeatedPtrField<ValueInfoProto> protos) {
+      for (auto& proto : protos) {
```

Comment:
const auto& since you are not modifying the proto

Reply:
+1